instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsAnnotationsInstrumentation.java

```diff
@@ -67,52 +68,62 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class JaxRsAnnotationsAdvice {
 
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void nameSpan(
-        @Advice.This Object target,
-        @Advice.Origin Method method,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelHandlerData") HandlerData handlerData,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-      callDepth = CallDepth.forClass(Path.class);
-      if (callDepth.getAndIncrement() > 0) {
-        return;
-      }
+    public static class AdviceScope {
+      private final HandlerData handlerData;
+      private final CallDepth callDepth;
+      private final Context context;
+      private final Scope scope;
+
+      public AdviceScope(CallDepth callDepth, Class<?> type, Method method) {
+        this.callDepth = callDepth;
+        if (callDepth.getAndIncrement() > 0) {
+          handlerData = null;
+          context = null;
+          scope = null;
+          return;
+        }
+
+        Context parentContext = Java8BytecodeBridge.currentContext();
```

Comment:
could just use `Context.current()` since this isn't in an advice any more

Reply:
Good point, I've applied this to other calls in `AdviceScope`.